mlflow/pyspark/ml/__init__.py

```diff
@@ -0,0 +1,285 @@
+import logging
+from pkg_resources import resource_filename
+
+import mlflow
+from mlflow.utils import chunk_dict, truncate_dict
+from mlflow.utils.annotations import experimental
+from mlflow.utils.autologging_utils import (
+    autologging_integration,
+    safe_patch,
+    try_mlflow_log,
+)
+
+
+_logger = logging.getLogger(__name__)
+
+
+FLAVOR_NAME = "pyspark.ml"
+
+
+def _get_fully_qualified_class_name(instance):
+    return instance.__class__.__module__ + "." + instance.__class__.__name__
+
+
+def _read_log_model_allowlist_from_file(allowlist_file):
+    allowlist = set()
+    with open(allowlist_file) as f:
+        for line in f:
+            stripped = line.strip()
+            is_blankline_or_comment = stripped == "" or stripped.startswith("#")
+            if not is_blankline_or_comment:
+                allowlist.add(stripped)
+    return allowlist
+
+
+def _read_log_model_allowlist(spark_session):
+    """
+    Reads the module allowlist and returns it as a set.
+    """
+    allowlist_file = spark_session.sparkContext._conf.get(
+        "spark.mlflow.pysparkml.autolog.logModelAllowlistFile", None
+    )
+    builtin_allowlist_file = resource_filename(__name__, "log_model_allowlist.txt")
+    if allowlist_file:
+        try:
+            return _read_log_model_allowlist_from_file(allowlist_file)
+        except Exception:
+            # fallback to built-in allowlist file
+            _logger.warning(
+                "Reading config spark.mlflow.pysparkml.autolog.logModelAllowlistFile "
+                f"{allowlist_file} failed, fallback to built-in allowlist file"
+            )
+            return _read_log_model_allowlist_from_file(builtin_allowlist_file)
+    else:
+        return _read_log_model_allowlist_from_file(builtin_allowlist_file)
+
+
+# When autolog called, `_log_model_allowlist` will be initialized.
+_log_model_allowlist = None
+
+
+def _get_warning_msg_for_skip_log_model(model):
+    return (
+        f"This model {model.uid} is not logged because it is not in "
+        "allowlist or its nested models are not in allowlist."
```

Comment:
```suggestion
        "or because one or more of its nested models are not allowlisted. Call mlflow.spark.log_model() to "
        "explicitly log the model, or specify a custom allowlist via the spark.mlflow.pysparkml.autolog.logModelAllowlistFile Spark conf (see mlflow.pyspark.ml.autolog docs for more info). 
```